File: src/main/java/frc/robot/teleopmodes/MainTeleop.java
package frc.robot.teleopmodes;

import org.wpilib.command3.Command;
import org.wpilib.command3.Scheduler;
import org.wpilib.opmode.PeriodicOpMode;
import org.wpilib.opmode.Teleop;

import frc.robot.Robot;

@Teleop
public class MainTeleop extends PeriodicOpMode {
    private final Robot m_robot;

    private final Command m_joystickDriveCommand;

    public MainTeleop(Robot robot) {
        m_robot = robot;
        var driverGamepad = robot.getDriverGamepad();
        m_joystickDriveCommand = m_robot.getDrive().getJoystickDriveCommand(driverGamepad.getHID());
    }

    @Override
    public void disabledPeriodic() {
        m_robot.robotPeriodic();
    }

    @Override
    public void start() {
        Scheduler.getDefault().schedule(m_joystickDriveCommand);

        var shooter = m_robot.getShooter();
        var driverGamepad = m_robot.getDriverGamepad();

        driverGamepad.rightBumper().and(driverGamepad.leftBumper()).whileTrue(shooter.getSpinAndFeedCommand());
        driverGamepad.rightBumper().and(driverGamepad.leftBumper().negate()).whileTrue(shooter.getSpinCommand());
    }

    @Override
    public void periodic() {
        m_robot.robotPeriodic();
    }

    @Override
    public void end() {
        Scheduler.getDefault().cancel(m_joystickDriveCommand);
    }

}

File: python/teleopmodes/main_teleop.py
# Copyright (c) FIRST and other WPILib contributors.
# Open Source Software; you can modify and/or share it under the terms of
# the WPILib BSD license file in the root directory of this project.

from commands3 import Command, Scheduler
from wpilib.opmode import PeriodicOpMode, teleop


@teleop
class MainTeleop(PeriodicOpMode):
    """Teleop mode with joystick drive and shooter control."""

    def __init__(self, robot):
        super().__init__()
        self._robot = robot
        driver_gamepad = robot.getDriverGamepad()
        self._joystick_drive_command = robot.getDrive().getJoystickDriveCommand(
            driver_gamepad.getHID()
        )

    def disabledPeriodic(self):
        self._robot.robotPeriodic()

    def start(self):
        Scheduler.getDefault().schedule(self._joystick_drive_command)

        shooter = self._robot.getShooter()
        driver_gamepad = self._robot.getDriverGamepad()

        (
            driver_gamepad.rightBumper()
            .and_(driver_gamepad.leftBumper())
            .whileTrue(shooter.getSpinAndFeedCommand())
        )
        (
            driver_gamepad.rightBumper()
            .and_(driver_gamepad.leftBumper().negate())
            .whileTrue(shooter.getSpinCommand())
        )

    def periodic(self):
        self._robot.robotPeriodic()

    def end(self):
        Scheduler.getDefault().cancel(self._joystick_drive_command)
